src/DssDirectDepositAaveDai.sol

```diff
@@ -216,25 +224,53 @@ contract DssDirectDepositAaveDai {
 
         emit File(what, data);
     }
+
     function file(bytes32 what, address data) external auth {
+        require(vat.live() == 1, "DssDirectDepositAaveDai/no-file-during-shutdown");
+
         if (what == "king") king = data;
         else revert("DssDirectDepositAaveDai/file-unrecognized-param");
         emit File(what, data);
     }
 
-    // --- Deposit controls ---
-    function wind(uint256 amount) external auth {
-        require(live == 1, "DssDirectDepositAaveDai/not-live");
+    // --- Automated Rate targeting ---
+    function calculateTargetSupply(uint256 targetInterestRate) public view returns (uint256) {
+        require(targetInterestRate > 0, "DssDirectDepositAaveDai/target-interest-zero");
+        require(targetInterestRate <= interestStrategy.getMaxVariableBorrowRate(), "DssDirectDepositAaveDai/above-max-interest");
 
-        _wind(amount);
+        // Do inverse calculation of interestStrategy
+        uint256 targetUtil;
+        if (targetInterestRate > _add(interestStrategy.baseVariableBorrowRate(), interestStrategy.variableRateSlope1())) {
+            // Excess interest rate
+            uint256 r = targetInterestRate - interestStrategy.baseVariableBorrowRate() - interestStrategy.variableRateSlope1();
+            targetUtil = _add(_rdiv(_rmul(interestStrategy.EXCESS_UTILIZATION_RATE(), r), interestStrategy.variableRateSlope2()), interestStrategy.OPTIMAL_UTILIZATION_RATE());
+        } else {
+            // Optimal interest rate
+            targetUtil = _rdiv(_rmul(_sub(targetInterestRate, interestStrategy.baseVariableBorrowRate()), interestStrategy.OPTIMAL_UTILIZATION_RATE()), interestStrategy.variableRateSlope1());
+        }
+        return _rdiv(_add(stableDebt.totalSupply(), variableDebt.totalSupply()), targetUtil);
     }
+
+    // --- Deposit controls ---
     function _wind(uint256 amount) internal {
+        bytes32 ilk_ = ilk;
+
+        // Wind amount is limited by the debt ceiling
+        (uint256 Art,,, uint256 line,) = vat.ilks(ilk_);
+        uint256 lineWad = line / RAY; // Round down to always be under the actual limit
+        uint256 newDebt = _add(Art, amount);
+        if (newDebt > lineWad) amount = _sub(lineWad, Art);
+
+        if (amount == 0) {
+            return;
```

Comment:
```suggestion
            emit Wind(amount);
            return;
```

Reply:
Hmm do we want to log this no-op? hmmm maybe yes but not 100% sure.

Reply:
It will make life easier for people writing scripts, less corner cases to consider as the function will either revert or emit.

Reply:
Same here, @hexonaut what do you think?

Reply:
Done.